lib/src/migration_visitor.dart

```diff
@@ -28,44 +28,48 @@ import 'utils.dart';
 /// new instance of itself (using [newInstance]) each time it encounters an
 /// `@import` or `@use` rule.
 abstract class MigrationVisitor extends RecursiveAstVisitor {
-  /// The stylesheet being migrated.
-  final Stylesheet stylesheet;
-
   /// A mapping from URLs to migrated contents for stylesheets already migrated.
-  final Map<Uri, String> migrated;
+  final _migrated = <Uri, String>{};
 
   /// True if dependencies should be migrated as well.
   final bool migrateDependencies;
 
   /// The patches to be applied to the stylesheet being migrated.
   UnmodifiableListView<Patch> get patches => UnmodifiableListView(_patches);
-  final List<Patch> _patches = [];
-
-  /// Constructs a new migration visitor, parsing the stylesheet at [url].
-  MigrationVisitor(Uri url, this.migrateDependencies,
-      {Map<Uri, String> migrated})
-      : stylesheet = parseStylesheet(url),
-        this.migrated = migrated ?? {};
+  List<Patch> _patches = [];
```

Comment:
```suggestion
  List<Patch> _patches;
```

The initial value isn't used anyway, since it's always reset in `visitStylesheet()`.